projboard/tests/test_article.py

```diff
@@ -0,0 +1,175 @@
+
+    # @pytest.mark.skip
+    @pytest.mark.django_db
+    def test_objectSavedInDb(self, create_articles):
+        assert create_articles[0] in Article.objects.all()
+        assert create_articles[1] in Article.objects.all()
+
+    # @pytest.mark.skip
+    @pytest.mark.django_db
+    # using Edite function
+    def test_EditObject(self, create_articles, create_subjects):
+
+        article = create_articles[0]
+        title = "New Title"
+        content = "New Content"
+        subject = create_subjects[1]
+        old_subject = create_subjects[0]
+
+        assert article.subject_id == old_subject
+
+        article.Edit(title, content, subject)
+
+        assert article.title == title
+        assert article.content == content
+        assert article.subject_id == subject
+        # if subject not in Article.SUBJECTS the value won't be changed
+        unsaved_subject = Subject(name='k')
+        article.Edit(subject=unsaved_subject)
+        assert article.subject_id == subject
+
+    # @pytest.mark.skip
+    @pytest.mark.django_db
+    def test_searchObjects(self, create_articles, create_subjects, create_user):
+        user1 = create_user[1]
+        article_list = [i for i in create_articles]
+        subject1 = create_subjects[1]
+
+        # search by title or a part of it
+        assert article_list[1] in Article.searchByTitle("Test")
+
+        # search by user
+        assert article_list[1] in Article.searchByUser(user1)
+
+        # search by subject
+        assert article_list[1] in Article.searchBySubject(subject1)
+
+    # @pytest.mark.skip
+    @pytest.mark.django_db
+    def test_searchObjectsNoneExists(self, create_articles, create_subjects, create_user):
+        # if not in db
+        user1 = create_user[1]
+        article_list = [i for i in create_articles]
+        subject1 = create_subjects[1]
+
+        # search by title or a part of it
+        assert article_list[1] not in Article.searchByTitle("Hakuna matata")
+
+        # search by user
+        assert article_list[0] not in Article.searchByUser(user1)
+
+        # search by subject
+        assert article_list[0] not in Article.searchBySubject(subject1)
+
+    @pytest.fixture
+    @pytest.mark.django_db
+    def create_like(self, create_user, create_articles):
+        like = Like(user_id=create_user[0], article_id=create_articles[0])
+        like.save()
+        return like
+
+    @pytest.fixture
+    @pytest.mark.django_db
+    def create_view(self, create_user, create_articles):
+        view = View(user_id=create_user[0], article_id=create_articles[0])
+        view.save()
+        return view
+
+    # @pytest.mark.skip
+    @pytest.mark.django_db
+    def test_filterByLikes(self):
+        # Filter By Likes
+        filteredArticles = Article.filterByLikes()
+        for i in range(len(filteredArticles) - 1):
+            print(filteredArticles)
+            if filteredArticles[i][1] < filteredArticles[i+1][1]:
```

Comment:
explaine:
filterByLikes() returns list of tuples => [ ( article object , num of likes ), .... ]
so im looping on the list and check if its not order from greates to least bynumlikes

Reply:
same as filterByViews()

Reply:
maybe you should order it in a better way 
https://github.com/redhat-beyond/Smarticle/pull/40/files#diff-b3a4ff9925d3172bbbc0154fc9ceef3131f5dda174217203439d651a5b7a0bc6R77 ? 

If you are doing gymnastic to get what you need, I assume you have to rework on it,
Dont you think?